plugins/modules/ldap_inc.py

```diff
@@ -0,0 +1,236 @@
+  - When implemented by the directory server, the module uses the ModifyIncrement extension
+    defined in RFC4525 and the control PostRead. This extension and the control are
+    implemented in OpenLdap but not all directory servers implement them. In this case, the
+    module automatically uses a more classic method based on two phases, first the current
+    value is read then the modify operation remove the old value and add the new one in a
+    single request. If the value has changed by a concurrent call then the remove action will
+    fail. Then the sequence is retried 3 times before raising an error to the playbook. In an
+    heavy modification environment, the module does not guarante to be systematically successful.
+  - This only deals with integer attribute of an existing entry. To modify attributes
+    of an entry, see M(community.general.ldap_attrs) or to add or remove whole entries,
+    see M(community.general.ldap_entry).
+  - The default authentication settings will attempt to use a SASL EXTERNAL
+    bind over a UNIX domain socket. If you need to use a simple bind to access
+    your server, pass the credentials in O(bind_dn) and O(bind_pw).
+author:
+  - Philippe Duveau (@pduveau)
+requirements:
+  - python-ldap
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: full
+options:
+  dn:
+    required: true
+    type: str
+    description:
+      - The DN entry containing the attribute to increment.
+  attribute:
+    required: true
+    type: str
+    description:
+      - The attribute to increment.
+  increment:
+    required: false
+    type: int
+    default: 1
+    description:
+      - The value of the increment to apply.
+  method:
+    required: false
+    type: str
+    default: auto
+    choices: [auto, rfc4525, legacy]
+    description:
+      - If 'auto', the module determines automatically the method to use.
+        If 'rfc4525' or 'legacy' force to use the corresponding method.
+extends_documentation_fragment:
+  - community.general.ldap.documentation
+  - community.general.attributes
+
+'''
+
+
+EXAMPLES = r'''
+- name: Increments uidNumber 1 Number for example.com
+  community.general.ldap_inc:
+    dn: "cn=uidNext,ou=unix-management,dc=example,dc=com"
+    attribute: "uidNumber"
+    increment: "1"
+  register: ldap_uidNumber_sequence
+
+- name: Modifies the user to define its identification number (uidNumber) when incrementation is successful.
+  community.general.ldap_attrs:
+    dn: "cn=john,ou=posix-users,dc=example,dc=com"
+    state: present
+    attributes:
+      - uidNumber: "{{ ldap_uidNumber_sequence.value }}"
+  when: ldap_uidNumber_sequence.incremented
+'''
+
+
+RETURN = r'''
+result:
+  description:
+    - attribute received the attributeType changed
+    - value receive the new value of the attribute in the specificied object
+  returned: success
+  type: int
+  sample:
+    - incremented: true
+    - attribute: "uidNumber"
+    - value: "2"
+    - rfc4525: true
```

Comment:
This should look as follows:
```suggestion
incremented:
  description:
    - ... some explanation what this means ...
  returned: success
  type: bool
  sample:
    - true

attribute:
  description:
    - The name of the attribute that was incremented.
  returned: success
  type: str
  sample: uidNumber

value:
  description:
    - The new value after incrementing.
  returned: success
  type: str
  sample: "2"

rfc4525:
  description:
    - ... some explanation what this means ...
  returned: success
  type: bool
  sample:
    - true
```